core/src/Streamly/Internal/Data/Either/Strict.hs

```diff
@@ -19,6 +19,8 @@
 --
 module Streamly.Internal.Data.Either.Strict
     ( Either' (..)
+    -- XXX Remove these and add lazyEither/strictEither to convert to and from
```

Comment:
Why remove these? We need to check for perf. Converting to lazy `Either` and checking adds another level of if/else which might be less performant.

Reply:
We can remove these accordingly.

Reply:
> Why remove these? We need to check for perf. Converting to lazy `Either` and checking adds another level of if/else which might be less performant.

It should get simplified by the compiler. We can check that.